### PR TITLE
[Re-release][PT-1.12.1] smdebug hooks fix,  smdebugger fix with numpy deprecation

### PR DESCRIPTION
**Pull request by ShiboXing**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ShiboXing** [comment]
sm test:
    efa: adc29ee
    standard: adc29ee
    rc: 179701f

File: pytorch/training/docker/1.12/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -129,8 +129,7 @@ RUN curl -L -o ~/mambaforge.sh https://github.com/conda-forge/miniforge/releases
     # Below 2 are included in miniconda base, but not mamba so need to install
     conda-content-trust \
     charset-normalizer \
-    # TMP FIX, NEED TO FIX SMDEBUG NUMPY DEPRECATION ERROR SOON: https://github.com/awslabs/sagemaker-debugger/blob/ff3b53607c36008164372b4fa785545f2e3ec2ef/smdebug/core/hook.py#L737
-    numpy=1.23.5 \
+    "shap<1" \
```
Comment (tejaschumbalkar) [inline pytorch/training/docker/1.12/py3/Dockerfile.cpu:132]: are we removing numpy?

File: pytorch/training/docker/1.12/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -278,8 +277,9 @@ RUN pip install --upgrade pip --trusted-host pypi.org --trusted-host files.pytho
 
 # Install smdebug from souce
 RUN cd /tmp \
-  && git clone -b ${SMDEBUG_VERSION} https://github.com/awslabs/sagemaker-debugger \
+  && git clone -b master --single-branch https://github.com/awslabs/sagemaker-debugger.git \
   && cd sagemaker-debugger \
+  && git checkout -f 9764a221fd3086ad8d2736128ccc2d69f269df1c \
```
Comment (tejaschumbalkar) [inline pytorch/training/docker/1.12/py3/Dockerfile.cpu:282]: what commit is this? can we add an explanation?

File: test/dlc_tests/container_tests/bin/testDependencyCheck [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
 apt install openjdk-8-jdk openjdk-8-jre -y
-cat >> /etc/environment <<EOL
+cat >> /etc/environment << EOL
 JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
 JRE_HOME=/usr/lib/jvm/java-8-openjdk-amd64/jre
 EOL
 
-pip install 'dependency-check<1'
-dependency-check --disableAssembly -s / -o build --exclude "**/dependency-check/**"
+export DEPENDENCY_CHECK_VERSION=7.4.4
```
Comment (tejaschumbalkar) [inline test/dlc_tests/container_tests/bin/testDependencyCheck:10]: lets parameterize depcheck version ref https://github.com/aws/deep-learning-containers/pull/2567/files

File: test/dlc_tests/container_tests/bin/testDependencyCheck [hunk 4]
```diff
@@ -1,11 +1,17 @@
...
+export DEPENDENCY_CHECK_URL=https://github.com/jeremylong/DependencyCheck/releases/download/v7.4.4/dependency-check-7.4.4-release.zip
+export DEPENDENCY_CHECK_HOME=./root/.local/dependency-check
```
Comment (tejaschumbalkar) [inline test/dlc_tests/container_tests/bin/testDependencyCheck:12]: do we need this env var?

File: pytorch/training/docker/1.12/py3/Dockerfile.cpu (hunk 2 above)
Comment (ShiboXing) [reply pytorch/training/docker/1.12/py3/Dockerfile.cpu:282]: added comment, it is the numpy deprecation fix

File: pytorch/training/docker/1.12/py3/Dockerfile.cpu (hunk 1 above)
Comment (ShiboXing) [reply pytorch/training/docker/1.12/py3/Dockerfile.cpu:132]: yes we will be using the latest one from conda since smdebug is fixed

**Review by tejaschumbalkar — APPROVED** [review]

**Pull request merged**